src/common/format/pretty-print.js

```diff
@@ -0,0 +1,56 @@
+import prettier from 'prettier/standalone';
+import markdown from 'prettier/parser-markdown';
+import unindent from 'strip-indent';
+
+const widths = { subject: 50, body: 72 };
+
+const config = { parser: 'markdown', plugins: [markdown] };
+
+function format(text, width) {
+  const options = { ...config, printWidth: width, proseWrap: 'always' };
+  return prettier.format(text, options);
+}
+
+function splitf(string, separator) {
+  const position = string.indexOf(separator);
+
+  if (position < 0) return [string];
+
+  const head = string.substring(0, position);
+  const tail = string.substring(position + separator.length);
+
+  return [head, tail];
+}
+
+function capitalizef(string) {
+  return string.replace(/[a-zA-Z]/, w => w.toUpperCase());
+}
+
+function gitsubject(string) {
```

Comment:
Also potentially interesting here: remove a dot at the end of the subject line _(the subject should not be a sentence)_

Reply:
Ah, I saw that in a post somewhere but didn't bother to include it here, assuming that most ticket descriptions aren't using a dot anyway 🤔 Do you think we should add that?

Reply:
Sure, but for the sake of consistency I feel like it still makes sense to add it. If we already opened the bucket of "Git Subject" then this is just a tiny extra on top.

Reply:
Cool! I'll add it when I have time 🙏 Thank you!

Reply:
Mmh, I thought I'd quickly add stripping of end-of-sentence punctuation marks for the Git subject lines… 🤔 

However, looking at it more closely it looks like a can of worms. To illustrate, here's a bunch of new tests I started out with:

```javascript
  it('strips trailing end-of-a-sentence punctuation marks', () => {
    expect(print('Strip full stops.')).toBe('Strip full stops');
    expect(print('Strip exclamation marks!')).toBe('Strip exclamation marks');
    expect(print('Strip question marks?')).toBe('Strip question marks');
    expect(print('Keep ellipsis...')).toBe('Keep ellipsis...');
    expect(print('Keep ellipsis…')).toBe('Keep ellipsis…');
  });
```

Also, I've had one existing test case that started failing and to me it's hard to handle such cases properly:

```javascript
  ● pretty-print › wraps overlong subject lines

    expect(value).toMatchSnapshot()

    Received value does not match stored snapshot "pretty-print wraps overlong subject lines 1".

    - Snapshot
    + Received

      "Wrap commit subject lines with more than 50…

      …characters and insert one blank line before the remaining subject text.
    - Wrap the remaining subject text to 72 characters.
    + Wrap the remaining subject text to 72 characters
      "

      17 |   it('wraps overlong subject lines', () => {
      18 |     const input = 'Wrap commit subject lines with more than 50 characters and insert one blank line before the remaining subject text. Wrap the remaining subject text to 72 characters.';
    > 19 |     expect(print(input)).toMatchSnapshot();
         |                          ^
      20 |   });
      21 |
      22 |   it('wraps overlong body lines', () => {

      at Object.toMatchSnapshot (src/common/format/pretty-print.test.js:19:26)
```

In this case, stripping the full stop just makes it look super-odd.

I'm currently leaning more towards leaving punctuation marks alone. What do you think @klappradla?

Reply:
@klappradla, this probably drowned between all the mails you received in your absence. ☝️
Maybe you can take another look? 🙏

Reply:
Sorry. I still have ~200 Github notification emails and wasn't brave enough to really tackle them yet...

I agree with what you wrote: let's not open this can of worms! If it's nothing that's done "easily" the naive way, then starting without it and looking at the "real world" usage makes more sense I'd say ✅

Reply:
Cool, thanks a lot @klappradla and I hope you'll make quick progress working through that impressive backlog of GitHub notifications 🤞😀